client/src/featureform/register.py

```diff
@@ -3947,6 +4010,194 @@ def search(self, raw_query, local=False):
             return search(processed_query, self._host)
 
 
+class ColumnResource:
+    """
+    Base class for all column resources. This class is not meant to be instantiated directly.
```

Comment:
> This class is not meant to be instantiated directly.

How about we make this an abstract class? https://www.geeksforgeeks.org/abstract-classes-in-python/

Reply:
I took a look at this as I was unfamiliar with the module, but I'm not sure it fits the use case here. I actually don't want to reimplement the methods in the derived classes, so having a functional base class is what I want, no?